src/main/java/com/algosenpai/app/stats/UserStats.java

```diff
@@ -116,15 +81,13 @@ public ChapterStat getStatsByIndex(int index) {
     }
 
     /**
-     * Update the stats for the current Chapter according to whether the user answered correctly or incorrectly.
-     * @param wasAnsweredCorrectly Whether the user answered correctly or incorrectly.
+     * TODO.
      */
-    public void updateCurrentChapter(boolean wasAnsweredCorrectly) {
-        currentChapter.totalAnswered++;
-        if (wasAnsweredCorrectly) {
-            currentChapter.correctAnswers++;
-        }
-        currentChapter.recalculateStats();
+    public void updateChapter(int index, int totalAnswered, int correct) {
```

Comment:
We r not using currentChapter anymore,  directly update the chapterData with a new batch of correctly and incorrect questions